Automattic/jetpack
# Masterbar: sync menu with latest WordPress.com changes

#### Changes proposed in this Pull Request:

I've noticed this recently and decided to handle it as part of HACK week cleanup.

After Calypso's sidebar has been redesigned, WordPress.com front end counterpart hasn't been updated to match it. Instead, it was simplified to just include top level items that link directly to Calypso. All of the sliding submenus have been removed.

However, we never followed up with similar change on Jetpack side which now causes weird functional and visual discrepancy for Jetpack and especially Atomic sites. On top of that, now it's completely out of sync with Calypso.

| Calypso | WP.com | Jetpack/Atomic |
| - | - | - |
|<img width="288" alt="Screenshot 2019-07-26 at 14 01 49" src="https://user-images.githubusercontent.com/1182160/61950395-0126c880-afae-11e9-89da-704883477e79.png">| <img width="328" alt="Screenshot 2019-07-26 at 13 53 51" src="https://user-images.githubusercontent.com/1182160/61950426-10a61180-afae-11e9-92a4-5fa5c3615fb9.png"> | <img width="302" alt="Screenshot 2019-07-26 at 13 58 31" src="https://user-images.githubusercontent.com/1182160/61950446-1c91d380-afae-11e9-9892-2f2c2fa9c4e6.png">|

I think we should just follow suit with WP.com and remove the submenus. In the long run, it would probably be best to completely replace it with something that can work on all platforms and be maintained in a single place (e.g. if we could move it to action bar or similar).

#### Testing instructions:

1. Use a test Jetpack site with working WordPress.com connection.
2. To enable WP.com masterbar, navigate to the bottom of `Settings -> Writing` section of you test site in Calypso and toggle `Enable the WordPress.com toolbar`.
3. On your Jetpack's site wp-admin and front end, click on various Masterbar top level links and verify that they take you directly to Calypso.
4. Sliding sidebars shouldn't appear anymore.
5. It should behave exactly like front end masterbar on WP.com.
